src/changefile/writeChangeFiles.ts

```diff
@@ -40,7 +40,9 @@ export function writeChangeFiles(
 
     stage(changeFiles, cwd);
     if (commitChangeFiles) {
-      commit('Change files', cwd);
+      // only commit change files, ignore other staged files/changes
+      const commitOptions = `${changeFiles.join(' ')}`;
```

Comment:
There are several things wrong with this:

1. The whole point of the `commit(msg, cwd, options)`'s new `options` param is to pass an array to the command. It should be just `const commitOptions = changeFiles;` according to the work done in `workspace-tools`.
2. Windows has an annoying problem of max length for a command: https://devblogs.microsoft.com/oldnewthing/20031210-00/?p=41553 - so it's a "bad idea" to shove a potentially growing list of changefiles through the command line like that.
3. You already staged the files with very specific files above. Essentially you're already doing what you intended to do. So I'm not sure if this change even solves the problems you're trying to solve. Have you tested your change before this PR?

Look at line 41-44:

```js
stage(changeFiles, cwd); // this line already staged files specifically for this commit: git add file1; git add file2; git add file3
if (commitChangeFiles) {
  commit("change files", cwd); // this line just calls git commit
}
```

In reading this your PR & your description and the workspace-tools change, I don't think you are solving what you intended.

Reply:
actually I suggest we maybe should do `const commitOptions = ['--only', '${workspaceRoot}/change/*.json'];`

Reply:
Good idea about `const commitOptions = ['--only', '${workspaceRoot}/change/*.json'];`

The goal of this PR is to make `yarn change` only commit the `change/*.json` files even if other files are staged.